### PR TITLE
Fix matrix exercise according to issue #170

### PR DESCRIPTION
**Pull request by pclausen**
Fix matrix exercise according to issue #170 

The row/columns where switched.

### CODE REVIEW

**Comment by zmoon** [comment]
Hi @pclausen. I think `r` and `c` of the stub (`matrix.f90`) are good to go with this change to the tests, but the `m` still needs its dimension and I would vote to remove the `A` declaration since it isn't part of the result or argument.

File: config.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "solution": [".meta/example.f90"],
     "test": ["%{snake_slug}_test.f90"],
     "example": ["%{snake_slug}.f90"],
-    "exemplar": ["%{snake_slug}.f90"]
+    "exemplar": ["exemplar.f90"]
```
Comment (SaschaMann) [inline config.json:17]: This change seems unrelated to the PR

**Review by SaschaMann — APPROVED** [review]
I can't really comment on the changes to the Fortran code but I'm gonna approve already so that you can merge the PR once @zmoon's changes have been addressed

File: config.json (hunk 1 above)
Comment (pclausen) [reply config.json:17]: Yes, you are right, but it came up as the linter has changed and I thought an extra PR was overkill... :-/

**Comment by zmoon** [comment]
> I think `r` and `c` of the stub (`matrix.f90`) are good to go with this change to the tests, but the `m` still needs its dimension and I would vote to remove the `A` declaration since it isn't part of the result or argument.

@pclausen, from what I can see, these haven't yet been addressed, `matrix.f90` doesn't seem to have been changed.

**Comment by pclausen** [comment]
@zmoon OK, I think I was a bit too fast and did not read your comments carefully enough. 

> but the m still needs its dimension

I guess I am not sure what you are hinting at here. Do you suggest a solution without giving `m` a dimension ie removing `m_dim`? I have not thought too much about it but it seems to complicate this exercise more than necessary but maybe that is because I am missing an obvious solution. 

Could you share what you think the solution should be to this problem ?

**Comment by zmoon** [comment]
Sorry, what I meant was that the `m` input has `character(len=*)` only in its declaration, it doesn't have `dimension(:)` like the example (`.meta/example.f90`) does, marking it as an array.

I don't suggest removing `m_dim`. I think that providing `m_dim` and providing `m` as an array of strings instead of a single string allows the problem to stay firmly in the "easy" category, which I think it is meant to be, having done this problem in other tracks.

And then my other comment was to remove to `A` declaration in `matrix.f90`. I think providing the declarations for the arguments and result only keeps it cleaner.
